OpenSim/Examples/DataTable/example.cpp

```diff
@@ -0,0 +1,357 @@
+    OpenSim::DataTable_<SimTK::Vec3> vec3{data_vec3.cbegin(),
+                                          data_vec3.cend(),
+                                          3,
+                                          OpenSim::COLWISE};
+    OpenSim::DataTable_<SimTK::Vec6> vec6{data_vec6.cbegin(),
+                                          data_vec6.cend(),
+                                          3,
+                                          OpenSim::COLWISE};
+  }
+
+  // Add multiple rows at once using an iterator. Because the DataTable is empty
+  // , number of columns has to specified when using iterator. When calling this
+  // function on a non-empty DataTable, the arugment ncol is ignored. We have 12
+  // elements in the iterator and following creates a 3x4 DataTable.
+  dt_real.addRows(data_real.cbegin(), data_real.cend(), 4);
+  dt_vec3.addRows(data_vec3.cbegin(), data_vec3.cend(), 4);
+  dt_vec6.addRows(data_vec6.cbegin(), data_vec6.cend(), 4);
+
+  std::vector<SimTK::Real> data1_real(4, 1);
+  std::vector<SimTK::Vec3> data1_vec3(4, {1, 2, 3});
+  std::vector<SimTK::Vec6> data1_vec6(4, {1, 2, 3, 4, 5, 6});
+
+  // Add a row to the DataTable using SimTK::RowVector_. Create SimTK::RowVector
+  // and add it as a row. After this the DataTable will of size 4x4.
+  SimTK::RowVector_<SimTK::Real> row_real{static_cast<int>(data1_real.size()), 
+                                          data1_real.data()};
+  SimTK::RowVector_<SimTK::Vec3> row_vec3{static_cast<int>(data1_vec3.size()), 
+                                          data1_vec3.data()};
+  SimTK::RowVector_<SimTK::Vec6> row_vec6{static_cast<int>(data1_vec6.size()), 
+                                          data1_vec6.data()};
+  dt_real.addRow(row_real);
+  dt_vec3.addRow(row_vec3);
+  dt_vec6.addRow(row_vec6);
+
+  // Add a single row to the DataTable using an iterator. Note that this 
+  // function is different from the one used above, which is addRows().
+  dt_real.addRow(data1_real.cbegin(), data1_real.cend());
+  dt_vec3.addRow(data1_vec3.cbegin(), data1_vec3.cend());
+  dt_vec6.addRow(data1_vec6.cbegin(), data1_vec6.cend());
+
+  // Trying to add a row using an iterator that does not produce enough 
+  // elements to fill up the row will throw as follows.
+  try {
+    dt_real.addRow(data1_real.cbegin(), data1_real.cend() - 1);
+  } catch(OpenSim::InvalidEntry&) {}
+  try {
+    dt_vec3.addRow(data1_vec3.cbegin(), data1_vec3.cend() - 1);
+  } catch(OpenSim::InvalidEntry&) {}
+  try {
+    dt_vec6.addRow(data1_vec6.cbegin(), data1_vec6.cend() - 1);
+  } catch(OpenSim::InvalidEntry&) {}
+
+  // DataTable can be told to allow for missing values when using iterator
+  // as follows. In this case, the missing values will be SimTK::NaN. See the
+  // documentation for addrow for details on the arguments. The third argument
+  // is ignored in this call.
+  dt_real.addRow(data1_real.cbegin(), data1_real.cend() - 1, 0, true);
```

Comment:
I thought of that but later I decided on this to allow more flexibility on per call basis. User can choose to use it differently on each call.